opentelemetry-api/src/logs/logger.rs

```diff
@@ -17,7 +17,7 @@ pub trait Logger {
 /// Interfaces that can create [`Logger`] instances.
 pub trait LoggerProvider {
     /// The [`Logger`] type that this provider will return.
-    type Logger: Logger;
```

Comment:
Hmm why remove this requirement here?

Reply:
Thanks for catching that! I was an accidental leftover I missed to remove.